File: algorithms/baekjoon/10811.js
const fs = require("fs");
const rawInput = fs.readFileSync("input").toString().trim(); // fs.readFileSync(0, "utf-8").trim()

const input = rawInput.split("\n").map((row) => row.split(" ").map(Number));
const basket = new Array(input[0][0]).fill().map((_, i) => i + 1);
const tasks = input.slice(1, input.length);

tasks.forEach(([i, j]) => {
  const targetRange = basket.slice(i - 1, j);
  targetRange.reverse();
  basket.splice(i - 1, j - i + 1, ...targetRange);
});

console.log(basket.join(" "));

File: algorithms/baekjoon/10811.py
# 제출 시 파일 입출력 부분 지우기

import sys

sys.stdin = open("input", "r")


def solution():
    N, M = map(int, input().split(" "))
    baskets = list(range(1, N + 1))
    for _ in range(M):
        i, j = map(int, input().split(" "))
        baskets[i - 1 : j] = reversed(baskets[i - 1 : j])
    print(" ".join(map(str, baskets)))


solution()
